pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerBuilderImpl.java

```diff
@@ -96,6 +96,9 @@ private ProducerBuilderImpl(PulsarClientImpl client, ProducerConfigurationData c
 
     @Override
     public CompletableFuture<Producer<T>> createAsync() {
+        // config validation
+        checkArgument(!(conf.isBatchingEnabled() && conf.isChunkingEnabled()),
```

Comment:
@rdhabalia The change looks good to me. About to enable the chunk, is it possible to add a config for the max chunk size and enable batching at the same time. If the message size is greater than the max chunk size and the chunk is enabled, we send messages with chunks, If the message is size is smaller than the max chunk size, we also can add them to a batch and if chunk is enabled, we make sure the match size can not exceed the max chunk size. So that it can be better applied when the user cannot confirm the message size for example from 100 bytes to 10MB. What do you think?

Reply:
@codelipenghui we can do but it will be little tricky to cover all usecases as we use batch-container if batching is enabled and combination of non-batch and batching will be little tricky. I have already put good amount of efforts to rebase and doing perf testing with changes. so, let's merge this PR and I will create a issue to add this feature and address it in separate PR as it will also require some effort to cover all unknowns.
So, can you please review this PR again as I am done with rebasing.